app/questionnaire/value_source_resolver.py

```diff
@@ -75,6 +75,23 @@ def _get_answer_value(
         ):
             return answer.value
 
+    def _resolve_list_item_id_for_answer(self, answer_id: str) -> str | None:
+        """
+        If there's a list item id and the answer is repeating, return the id to resolve the instance of the answer
+        However if the answer is repeating for a different list, return None so that the repeating answer id resolves to a list
+        """
+        if self.list_item_id and (
+            answer_list := self.schema.get_list_name_for_answer_id(answer_id)
+        ):
+            # if there is a current list, and it differs to the repeating answer one, return None
+            if (
+                self.location
+                and self.location.list_name
+                and self.location.list_name != answer_list
+            ):
+                return None
+            return self.list_item_id
```

Comment:
Just a detail. It, sort of, suggests that the var is a list of answers, when it's a list name, so maybe `list_name`, `list_name_for_answer` or `list_name_anwer_id` or something similar?
```suggestion
        if self.list_item_id and (
            list_name_for_answer := self.schema.get_list_name_for_answer_id(answer_id)
        ):
            # if there is a current list, and it differs to the repeating answer one, return None
            if (
                self.location
                and self.location.list_name
                and self.location.list_name != list_name_for_answer
            ):
                return None
            return self.list_item_id
```